src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicBuffers.cs

```diff
@@ -0,0 +1,91 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using Microsoft.Quic;
+
+namespace System.Net.Quic.Implementations.MsQuic.Internal;
+
+/// <summary>
+/// Helper class to convert managed data into QUIC_BUFFER* consumable by MsQuic.
+/// It also allows reuse with repeated Reset/Initialize calls.
+/// Note that since this is struct and there's no finalizer, Dispose must be always called to release the unmanaged memory allocated by this struct.
+/// </summary>
+internal unsafe struct MsQuicBuffers : IDisposable
+{
+    // Handles to pinned memory blocks from the user.
+    private MemoryHandle[] _handles;
+    // Native memory block which holds the pinned memory pointers from _handles and can be passed to MsQuic as QUIC_BUFFER*.
+    private void* _buffers;
```

Comment:
```suggestion
    private QUIC_BUFFER* _buffers;
```
Also, add cast around allocation and remove the cast from `Buffers` property

Reply:
Done.